src/main/java/Duke.java

```diff
@@ -1,103 +1,222 @@
+            "\t  mark (no.) : marks task number (no.) as completed.\n" +
+            "\t  unmark (no.) : marks task number (no.) as incomplete.\n" +
+            "\t  todo (task) : adds task with description (task) to the tasklist.\n" +
+            "\t  deadline (task) /by (date) : adds task with description (task) and due date (date) to the tasklist.\n" +
+            "\t  event (task) /from (date1) /to (date2) : adds task with description (task) " +
+            "and start date (date1) and end date (date2)\n" +
+            "\t  help : shows list of available commands" +
+            LINE_PARTITION;
+
+    public static void main(String[] args) {
+        System.out.print(GREET);
+        while(true) {
+            takeUserInput();
+        }
     }
-    public static void printTaskNotFound(int taskNum) {
-        System.out.print(LINE_PARTITION +
-                "\t  Task number " + taskNum + " does not exist, please try again.\n" +
-                LINE_PARTITION);
+
+    public static void takeUserInput() {
+        final String userInput = in.nextLine();
+        final String[] cmdTypeAndParams = splitUserInput(userInput);
+        final String cmd = cmdTypeAndParams[0];
+        final String args = cmdTypeAndParams[1];
+
+        switch (cmd) {
+        case "help" :
+            printHelp();
+        case "list":
+            executeList();
+            break;
+        case "mark":
+            executeMark(args, true);
+            break;
+        case "unmark":
+            executeMark(args, false);
+            break;
+        case "todo":
+            executeTaskAdd(args, "todo");
+            break;
+        case "deadline":
+            executeTaskAdd(args, "deadline");
+            break;
+        case "event":
+            executeTaskAdd(args, "event");
+            break;
+        case "bye":
+            executeBye();
+        default:
+            invalidCmd();
+        }
     }
-    public static void printMark(Task[] taskList, int number) {
-        taskList[number].setDone(true);
-        System.out.print(LINE_PARTITION +
-                "\t  Great job! I will mark this task as done: \n" +
-                "\t  [X] " + taskList[number].getName() + "\n" +
-                LINE_PARTITION);
+
+    private static void printHelp() {
+        System.out.print(HELPLIST);
     }
-    public static void printUnmark(Task[] taskList, int taskNum) {
-        taskList[taskNum].setDone(false);
-        System.out.print(LINE_PARTITION +
-                "\t  Alright, I have marked this task as undone: \n" +
-                "\t  [ ] " + taskList[taskNum].getName() + "\n" +
-                LINE_PARTITION);
+    private static String[] splitUserInput(String rawUserInput) {
+        final String[] split = rawUserInput.trim().split("\\s+", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""}; // else case: no parameters
```

Comment:
Maybe you could move the comment along the same indentation so as to avoid breaking the logical structure of the program.